src/mypy_zope/plugin.py

```diff
@@ -258,17 +259,21 @@ def get_metaclass_hook(
         self, fullname: str
     ) -> Optional[Callable[[ClassDefContext], None]]:
         # print(f"get_metaclass_hook: {fullname}")
-        return None
+        def analyze_metaclass(ctx: ClassDefContext) -> None:
+            metaclass = cast(NameExpr, ctx.cls.metaclass)
+            info = cast(TypeInfo, metaclass.node)
+            expected = "zope.interface.interface.InterfaceClass"
```

Comment:
Is it possible to avoid calling `analyze_metaclass` here by inspecting `fullname` parameter?

Reply:
Fullname is an arbitrary name, like 'foo.bar.Baz'. I don't see how that reveals anything about the metaclass of Baz, something that can only be done once one has a `ctx`. At least, that's how I understand the static analysis constraints from what I've seen of it (which isn't that much). If you have a better idea, I'd be happy to explore it.

Reply:
Ah, I see, that makes sense. I just wanted to make sure we are not missing this.